src/vs/server/node/remoteAgentEnvironmentImpl.ts

```diff
@@ -349,4 +353,40 @@ export class RemoteAgentEnvironmentChannel implements IServerChannel {
 		return scannedExtension ? toExtensionDescription(scannedExtension, false) : null;
 	}
 
+	private async _ensureLanguagePackIsInstalled(language: string): Promise<void> {
+		if (
+			// No need to install language packs for the default language
+			language === platform.LANGUAGE_DEFAULT ||
+			// The extension gallery service needs to be available
+			!this._extensionGalleryService.isEnabled()
+		) {
+			return;
+		}
+		const installed = await this._languagePackService.getInstalledLanguages();
+		if (installed.find(p => p.id === language)) {
+			// language pack already installed.
+			return;
+		}
+		try {
+			const tagResult = await this._extensionGalleryService.query({ text: `tag:lp-${language}` }, CancellationToken.None);
+			if (tagResult.total === 0) {
+				this._logService.warn(`No extension found for language ${language}. This is unexpected since the client is configured to use this language.`);
+				return;
+			}
+
+			const extensionToInstall = tagResult.total === 1
+				? tagResult.firstPage[0]
+				// If there are multiple extensions, prefer the one from Microsoft. This mirrors the client behavior. This "multiple extensions" case hasn't
+				// come up yet, but it is possible that a language pack is published by a third party.
+				: tagResult.firstPage.find(e => e.publisher === 'MS-CEINTL' && e.name.startsWith('vscode-language-pack')) ?? tagResult.firstPage[0];
+			this._logService.trace(`Language Pack for ${language} is not installed. It will be installed now.`);
+			await this._extensionManagementCLI.installExtensions([extensionToInstall.identifier.id], [], { isMachineScoped: true }, true, {
+				log: (s) => this._logService.info(s),
+				error: (s) => this._logService.error(s)
+			});
```

Comment:
We should not install extensions without explicit user consent unless they come from an allow-list. Otherwise, this would introduce a security vulnerability.